erpnext/public/js/controllers/transaction.js

```diff
@@ -183,6 +183,9 @@ erpnext.TransactionController = erpnext.taxes_and_totals.extend({
 				() => set_value('price_list_currency', currency),
 				() => set_value('status', 'Draft'),
 				() => set_value('is_subcontracted', 'No'),
+				() => {
+					if(this.frm.doc.party_name) { return this.frm.trigger('party_name'); } // pricelist selection
```

Comment:
party_name field only exists in the quotation form and not in any other sales/purchases doctypes. So instead of adding this code in transaction.js file you can add it in the quotation.js file.